### PR TITLE
Implement triangle and octagon outputs

### PR DESCRIPTION
**Pull request by julienf-unity**
### Purpose of this PR
Implement triangle and octagon outputs

---
### Testing status
**Katana Tests**:
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=vfx%2Ffeature%2Ftriangle-primitive&unity_branch=trunk&automation-tools_branch=master

**Manual Tests**: Yes

**Automated Tests**: New graphics test

---
### Overall Product Risks
**Technical Risk**: 1
**Halo Effect**: 1

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: TestProjects/VisualEffectGraph/Assets/AllTests/Editor/Tests/VFXSpaceBoundTest.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿#if !UNITY_EDITOR_OSX || MAC_FORCE_TESTS
+#if !UNITY_EDITOR_OSX || MAC_FORCE_TESTS
```
Comment (PaulDemeulenaere) [inline TestProjects/VisualEffectGraph/Assets/AllTests/Editor/Tests/VFXSpaceBoundTest.cs]: Just a remark : I still don't know what is this often reported changed in our PR...

File: com.unity.visualeffectgraph/Shaders/ParticlePlanarPrimitives/PassGS.template [hunk 2]
```diff
@@ -0,0 +1,103 @@
+#pragma require geometry
+#pragma vertex vert
```
Comment (PaulDemeulenaere) [inline com.unity.visualeffectgraph/Shaders/ParticlePlanarPrimitives/PassGS.template]: Why rename hasn't been detected on this MR ? :-/
It was Shaders/ParticleQuads/PassGS.template before, isn't it ?

File: com.unity.visualeffectgraph/Shaders/VFXFillIndirectArgs.compute [hunk 3]
```diff
@@ -73,45 +70,69 @@ void CSVFXIndirectArgs(uint3 id : SV_DispatchThreadID)
...
+            case 5: // triangles
+            {
+                args.x = elementCount * 3;
+                args.y = 1;
+                break;
+            }
+			
+			case 6: // octagon
+			{
+                if (elementCount > 8192) // Due to 16bits indices (8 vertices per element)
```
Comment (PaulDemeulenaere) [inline com.unity.visualeffectgraph/Shaders/VFXFillIndirectArgs.compute]: You have mixed space and tab in this file

**Comment by julienf-unity** [comment]
Katana is green

**Comment by julienf-unity** [comment]
Katana is green. Can this be merge bow that I finally manage to get a window with a green katana !

**Comment by andybak** [comment]
I was hoping the addition of triangle primitives might allow rendering of meshes via the VFX Graph. I'm thinking of applications like destruction/assembly and complex procedural mesh animation.

Unfortunately - the triangles are all equilateral and it's therefore impossible to represent arbitrary meshes. Is there any point opening an feature request for vertex level access to triangle primitives?

**Pull request merged**